src/protocols/UBX.hpp

```diff
@@ -0,0 +1,134 @@
+#ifndef GPS_UBLOX_PROTOCOLS_UBX_HPP
+#define GPS_UBLOX_PROTOCOLS_UBX_HPP
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <vector>
+
+#include <gps_ublox/protocols/Parser.hpp>
+
+namespace gps_ublox
+{
+    namespace protocols {
+        class UBX : public Parser
+        {
+            private:
+                /** Computes the 2-byte checksum over the given buffer
+                 */
+                static std::vector<uint8_t> checksum(const uint8_t *buffer, const uint8_t *end);
+
+            public:
+                static const uint8_t SYNC_1;
+                static const uint8_t SYNC_2;
+                static const uint8_t SYNC_1_IDX;
+                static const uint8_t SYNC_2_IDX;
+                static const uint8_t MSG_CLASS_IDX;
+                static const uint8_t MSG_ID_IDX;
+                static const uint8_t LENGTH_LSB_IDX; // the payload-only length (LSB)
+                static const uint8_t LENGTH_MSB_IDX; // the payload-only length (MSB)
+                static const uint8_t PAYLOAD_IDX;
+                static const size_t FRAMING_SIZE_OVERHEAD;
+
+                /** The message class enumeration
+                 */
+                enum MsgClass {
+                    UBX_ACK = 0x05,
+                    UBX_CFG = 0x06
+                };
+
+                /** The message id enumeration
+                 */
+                enum MsgId {
+                    NACK = 0x00,
+                    ACK = 0x01,
+                    VALSET = 0x8A
+                };
+
+                /** The layer where a configuration data is stored
+                 */
+                enum ConfigurationLayer {
+                    LAYER_RAM = 1,
+                    LAYER_BBR = 2,
+                    LAYER_FLASH = 4,
+                    LAYER_ALL = 7
+                };
+
+                /** The unique key id of a configuration value
+                 */
+                enum ConfigurationKeyId {
```

Comment:
Same ... would really try to use the same wording everywhere for a given word. Short or not short.

Since Configuration is very widely used, I would personally prefer `Config` or the full `Configuration`.